home/tests/test_models.py

```diff
@@ -225,16 +250,41 @@ def test_template_not_submitted_with_no_message(
         """
         home_page = HomePage.objects.first()
         main_menu = PageBuilder.build_cpi(home_page, "main-menu", "Main Menu")
-        PageBuilder.build_cp(
-            parent=main_menu,
-            slug="ha-menu",
-            title="HealthAlert menu",
-            bodies=[],
-            whatsapp_template_name="WA_Title_1",
-        )
+        with self.assertRaises(ValidationError):
+            PageBuilder.build_cp(
+                parent=main_menu,
+                slug="ha-menu",
+                title="HealthAlert menu",
+                bodies=[],
+                whatsapp_template_name="WA_Title_1",
+            )
 
         mock_create_whatsapp_template.assert_not_called()
 
+    def test_clean_text_valid_variables(self):
+        """
+        The message should accept variables if and only if they are numeric and ordered
+        """
+        home_page = HomePage.objects.first()
+        main_menu = PageBuilder.build_cpi(home_page, "main-menu", "Main Menu")
+        with self.assertRaises(ValidationError):
+            PageBuilder.build_cp(
+                parent=main_menu,
+                slug="ha-menu",
+                title="HealthAlert menu",
+                bodies=[
+                    WABody(
+                        "WA Title",
+                        [
+                            WABlk(
+                                "{{2}}{{1}} {{foo}}",
+                            )
+                        ],
+                    )
+                ],
+                whatsapp_template_name="WA_Title_1",
+            )
+
```

Comment:
I think we also need a test for the WhatsApp title validation, to ensure that if there's no whatsapp title, then the ValidationError is raised to send the error message to the user